dags/dap_collector_ppa_prod.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+
+from airflow import DAG
+
+from operators.gcp_container_operator import GKEPodOperator
+from utils.tags import Tag
+
+DOCS = """
+### PPA Prod DAP Collector
+
+#### Description
+
+Runs a Docker image that collects PPA Prod Environment data from a DAP (Distributed Aggregation Protocol) leader and stores it in BigQuery.
+
+The container is defined in
+[docker-etl](https://github.com/mozilla/docker-etl/tree/main/jobs/dap-collector-ppa-prod)
+
+This DAG requires following variables to be defined in Airflow:
+* dap_ppa_prod_auth_token
+* dap_ppa_prod_hpke_private_key
+* dap_ppa_prod_task_config_url
+* dap_ppa_prod_ad_config_url
+
+This job is under active development, occasional failures are expected.
+
+#### Owner
+
+bbirdsong@mozilla.com
+"""
+
+default_args = {
+    "owner": "bbirdsong@mozilla.com",
+    "email": ["ads-eng@mozilla.com", "bbirdsong@mozilla.com"],
+    "depends_on_past": False,
+    "start_date": datetime(2024, 6, 26),
+    "email_on_failure": True,
+    "email_on_retry": False,
+    "retries": 0,
+}
+
+project_id = "moz-fx-ads-prod"
+ad_table_id = "ppa.measurements"
+report_table_id = "ppa.reports"
+
+tags = [
+    Tag.ImpactTier.tier_3,
+    Tag.Triage.no_triage,
+]
+
+
+with DAG(
+    "dap_collector_ppa_prod",
+    default_args=default_args,
+    doc_md=DOCS,
+    schedule_interval="*/15 * * * *",
+    tags=tags,
+    catchup=False,
+) as dag:
+    dap_collector = GKEPodOperator(
+        task_id="dap_collector_ppa_prod",
+        arguments=[
+            "python",
+            "dap_collector_ppa_prod/main.py",
+            "--date={{ ts }}",
+            "--auth-token={{ var.value.dap_ppa_prod_auth_token }}",
+            "--hpke-private-key={{ var.value.dap_ppa_prod_hpke_private_key }}",
```

Comment:
same question about secrets as the token

Reply:
yeah the private key and auth token are credentials for the dap collector to be able to read data from the leader.

Reading through https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/27922811/WTMO+Developer+Guide#WTMODeveloperGuide-SecretManagement it looks like it should be a simple adjustment in this file to read from secrets. I'll need help setting the values in the secrets though

Reply:
I updated the secrets to be read from GSM, will coordinate with you to ensure the names of the values match. Also updated the docker-etl job to get these from environment variables instead of arguments https://github.com/mozilla/docker-etl/pull/219